src/coprocessor/dag/batch/statistics.rs

```diff
@@ -69,18 +69,14 @@ pub trait ExecSummaryCollector: Send {
     where
         Self: Sized;
 
-    /// Returns an instance that will record elapsed duration. The instance should be later
-    /// passed back to `inc_elapsed_duration` when processing is complete.
-    fn start_record_duration(&self) -> Self::DurationRecorder;
+    /// Returns an instance that will record elapsed duration and increase
+    /// the iterations counter. The instance should be later passed back to
+    ///  `finish_batch` when processing of `next_batch` is completed.
+    fn start_batch(&mut self) -> Self::DurationRecorder;
```

Comment:
sorry for the late comment. I think adding `on_` prefix should be better, because this function does not start batch actually. Instead it does something when batch is started.